src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -509,15 +509,20 @@ private AIState lookForRequests()
                 {
                     //Seems like somebody else picked up our stack.
                     //Lets try this again.
-                    if (!async)
+                    if (async)
                     {
-                        worker.getCitizenData().createRequest(firstDeliverableRequest.getRequest());
+                        worker.getCitizenData().createRequestAsync(firstDeliverableRequest.getRequest());
                     }
                     else
                     {
-                        worker.getCitizenData().createRequestAsync(firstDeliverableRequest.getRequest());
+                        worker.getCitizenData().createRequest(firstDeliverableRequest.getRequest());
                     }
                 }
+
+                if(async)
```

Comment:
No. This changes changes critical behaviour. You need to change the initial if switch to exclude asyncs. See the AITarget on how to check if any none async requests are open.

Reply:
kk